### PR TITLE
CachedInterpreter: New kind of jit which always fallback to interpreter.

### PR DESCRIPTION
**Pull request by degasus**
It's twice as fast as the interpreter, without any assembly.

### CODE REVIEW

**Comment by endrift** [comment]
Calling this a JIT is confusing: unless I'm mistaken, it does no just-in-time recompliation, only instruction sequence caching. Perhaps then it should be called a caching interpreter? I've seen other emulators use that phrase before; it might be less confusing that way.

**Comment by delroth** [comment]
Definitely not a JIT indeed. The fact that the basic block caching implementation is currently part of JitCommon is an irrelevant detail, and imo it's worth fixing.

File: Source/Core/DolphinWX/Config/GeneralConfigPane.cpp [hunk 1]
```diff
@@ -22,12 +22,13 @@ GeneralConfigPane::GeneralConfigPane(wxWindow* parent, wxWindowID id)
...
 {
 	cpu_cores = {
-		{ 0, _("Interpreter (VERY slow)") },
+		{ PowerPC::CORE_INTERPRETER, _("Interpreter (VERY slow)") },
+		{ PowerPC::CORE_JITGENERIC, _("JIT Generic (slow)") },
 #ifdef _M_X86_64
-		{ 1, _("JIT Recompiler (recommended)") },
-		{ 2, _("JITIL Recompiler (slower, experimental)") },
+		{ PowerPC::CORE_JIT64, _("JIT Recompiler (recommended)") },
+		{ PowerPC::CORE_JITIL64, _("JITIL Recompiler (slower, experimental)") },
```
Comment (JosJuice) [inline Source/Core/DolphinWX/Config/GeneralConfigPane.cpp]: It's misleading that JITIL says "slower" while the option that this PR adds says "slow". Could we make JIT say "fastest, recommended" and JITIL only "experimental"?

**Comment by lioncash** [comment]
Perhaps this should be named something like "Cached Interpreter" or something, considering it doesn't actually act as a JIT.

File: Source/Core/DolphinWX/Config/GeneralConfigPane.cpp (hunk 1 above)
Comment (MoochMcGee) [inline Source/Core/DolphinWX/Config/GeneralConfigPane.cpp]: I second this notion.

**Comment by delroth** [comment]
It's also fairly close to what is generally called subroutine threading: https://en.wikipedia.org/wiki/Threaded_code#Subroutine_threading

(though this doesn't generate a block that calls functions -- it just pre-decodes.)

**Comment by randomstuff** [comment]
I'd say this is Direct Threaded Code with [Call Threading](http://www.complang.tuwien.ac.at/forth/threaded-code.html#call-threading). Subroutine threaded would JIT a sequence of call instruction for each block.

**Comment by degasus** [comment]
Renamed to "CachedInterpreter". The name "threading xxx" sounds too confusing for users to me, it's not multithreaded.

**Comment by JMC47** [comment]
I tested this and in CPU intensive games, it was about 3x faster than the regular interpreter.

**Comment by degasus** [comment]
So everything is fine and there is no remaining criticism?

Ready to merge :D

**Comment by JMC47** [comment]
I'm good with this, we need more stuff for the progress report.

**Pull request merged**